server/admin.api.js

```diff
@@ -66,23 +70,19 @@ router.post('/admin/cleanup', mw.adminOnly, (req, res)=>{
 });
 
 /* Searches for matching edit or share id, also attempts to partial match */
-router.get('/admin/lookup/:id', mw.adminOnly, async (req, res, next)=>{
-	HomebrewModel.findOne({
-		$or : [
-			{ editId: { $regex: req.params.id, $options: 'i' } },
-			{ shareId: { $regex: req.params.id, $options: 'i' } },
-		]
-	}).exec()
-	.then((brew)=>{
-		if(!brew)	// No document found
+router.get('/admin/lookup/:id', mw.adminOnly, asyncHandler(HomebrewAPI.getBrew('admin', false)), async (req, res, next)=>{
+	const brew = req?.brew ?? undefined;
+
+	try {
```

Comment:
This `try` block should no longer be necessary since we have removed the database/Google action inside that could potentially fail. Instead we use `getBrew()` which has its own error handling inside.

If there is any issue getting a brew, `getBrew()` already generates and throws the appropriate error, and then `asyncHandler` passes the error to the central error handler in `app.js` which knows to send the 404 to the client.

So if we found the brew, we can just return the brew. If there was an error, we won't reach this line anyway.

```
router.get('/admin/lookup/:id', mw.adminOnly, asyncHandler(HomebrewAPI.getBrew('admin', false)), async (req, res, next)=>{
		return res.json(brew);
});
```

Reply:
Applied in the latest commit.